src/gsy_e/gsy_e_core/redis_connections/redis_area_market_communicator.py

```diff
@@ -15,86 +15,136 @@
+from typing import Dict
+from collections.abc import Callable
+
+from gsy_framework.constants_limits import ConstSettings
+from redis import StrictRedis
+from rq import Queue
+
 import gsy_e.constants
+from gsy_e.constants import REDIS_PUBLISH_RESPONSE_TIMEOUT
+from gsy_e.gsy_e_core.redis_connections.aggregator_connection import AggregatorHandler
+from gsy_e.gsy_e_core.redis_connections.redis_communication import REDIS_URL
 
 log = logging.getLogger(__name__)
 REDIS_THREAD_JOIN_TIMEOUT = 2
 REDIS_POLL_TIMEOUT = 0.01
 
 
 class RedisCommunicator:
+    """Base class for redis communication using pubsub."""
+
     def __init__(self):
         self.redis_db = StrictRedis.from_url(REDIS_URL, retry_on_timeout=True)
         self.pubsub = self.redis_db.pubsub()
         self.pubsub_response = self.redis_db.pubsub()
         self.event = Event()
 
-    def publish(self, channel, data):
+    def publish(self, channel: str, data: str):
+        """Publish message on redis channel."""
         self.redis_db.publish(channel, data)
 
     def wait(self):
+        """Wait for thread event to be performed."""
         self.event.wait()
         self.event.clear()
 
     def resume(self):
+        """Resume receiving events."""
         self.event.set()
 
-    def sub_to_response(self, channel, callback):
+    def sub_to_response(self, channel: str, callback: Callable) -> Thread:
+        """Subscribe to response channel and return pubsub thread."""
         self.pubsub_response.subscribe(**{channel: callback})
         thread = self.pubsub_response.run_in_thread(daemon=True)
         log.trace(f"Started thread for responses: {thread}")
         return thread
 
-    def sub_to_channel(self, channel, callback):
+    def sub_to_channel(self, channel: str, callback: Callable) -> Thread:
+        """Subscribe to channel and return pubsub thread."""
         self.pubsub.subscribe(**{channel: callback})
         thread = self.pubsub.run_in_thread(daemon=True)
         log.trace(f"Started thread for events: {thread}")
         return thread
 
 
+class BlockingCommunicator(RedisCommunicator):
+    """Communicator for sending blocking messages via redis pubsub."""
+
+    def __init__(self):
+        super().__init__()
+        self.lock = Lock()
+
+    def sub_to_channel(self, channel: str, callback: Callable):
+        """Subscribe to channel."""
+        self.pubsub.subscribe(**{channel: callback})
+
+    def poll_until_response_received(self, response_received_callback: Callable):
+        """Wait until response of send message was received."""
+        start_time = time()
+        while not response_received_callback() and \
+                (time() - start_time < REDIS_PUBLISH_RESPONSE_TIMEOUT):
+            with self.lock:
+                self.pubsub.get_message(timeout=REDIS_POLL_TIMEOUT)
+
+
 class ResettableCommunicator(RedisCommunicator):
+    """Communicator for sending messages using redis pubsub including."""
```

Comment:
Should "including" be followed by something? Also, could you please specify in the docstring what "resettable" means?

Reply:
https://github.com/gridsingularity/gsy-e/pull/1366/commits/66b747ac0df704e94b3f2d895a279ed233b2bad2